tests/node/test_wakunode_sharding.nim

```diff
@@ -0,0 +1,1026 @@
+      assertResultOk(await serverHandler1.waitForResult(FUTURE_TIMEOUT))
+      assertResultOk(await clientHandler1.waitForResult(FUTURE_TIMEOUT))
+      check:
+        (await serverHandler2.waitForResult(FUTURE_TIMEOUT)).isErr()
+        (await clientHandler2.waitForResult(FUTURE_TIMEOUT)).isErr()
+
+      # When the client publishes a message in contentTopic2
+      serverHandler1.reset()
+      serverHandler2.reset()
+      clientHandler1.reset()
+      clientHandler2.reset()
+      discard await client.publish(
+        some(pubsubTopic2),
+        WakuMessage(payload: "message2".toBytes(), contentTopic: contentTopic2),
+      )
+
+      # Then the server and client receive the message in contentTopic2's handlers, but not in contentTopic1's
+      assertResultOk(await serverHandler2.waitForResult(FUTURE_TIMEOUT))
+      assertResultOk(await clientHandler2.waitForResult(FUTURE_TIMEOUT))
+      check:
+        (await serverHandler1.waitForResult(FUTURE_TIMEOUT)).isErr()
+        (await clientHandler1.waitForResult(FUTURE_TIMEOUT)).isErr()
+
+    asyncTest "Configure Node combining Multiple Pubsub and Content Topics":
+      # Given a connected server and client subscribed to multiple pubsub topics and content topics
+      let
+        contentTopic = "myContentTopic"
+        pubsubTopic1 = "/waku/2/rs/0/1"
+        pubsubTopic2 = "/waku/2/rs/0/2"
+        serverHandler1 = server.subscribeCompletionHandler(pubsubTopic1)
+        clientHandler1 = client.subscribeCompletionHandler(pubsubTopic1)
+        serverHandler2 = server.subscribeCompletionHandler(pubsubTopic2)
+        clientHandler2 = client.subscribeCompletionHandler(pubsubTopic2)
+        contentTopic3 = "/toychat/2/huilong/proto"
+        pubsubTopic3 = "/waku/2/rs/0/58355"
+          # Automatically generated from the contentTopic above
+        contentTopic4 = "/0/toychat2/2/huilong/proto"
+        pubsubTopic4 = "/waku/2/rs/0/23286"
+          # Automatically generated from the contentTopic above
+        serverHandler3 = server.subscribeToContentTopicWithHandler(contentTopic3)
+        clientHandler3 = client.subscribeToContentTopicWithHandler(contentTopic3)
+        serverHandler4 = server.subscribeToContentTopicWithHandler(contentTopic4)
+        clientHandler4 = client.subscribeToContentTopicWithHandler(contentTopic4)
+
+      await client.connectToNodes(@[server.switch.peerInfo.toRemotePeerInfo()])
+
+      # When the client publishes a message in the topic1
+      discard await client.publish(
+        some(pubsubTopic1),
+        WakuMessage(payload: "message1".toBytes(), contentTopic: contentTopic),
+      )
+
+      # Then the server and client receive the message in topic1's handlers, but not in topic234's
+      assertResultOk(await serverHandler1.waitForResult(FUTURE_TIMEOUT))
+      assertResultOk(await clientHandler1.waitForResult(FUTURE_TIMEOUT))
+      check:
+        (await serverHandler2.waitForResult(FUTURE_TIMEOUT)).isErr()
```

Comment:
If I'm not mistaken it is 6 sec wait added here. Can it be shorter or wait parallel somehow and than check ?